src/utils/claimRewards.js

```diff
@@ -0,0 +1,231 @@
+  toBuffer(rest).copy(buf, 32);
+  const dataBuf = keccak256(buf);
+
+  buf = Buffer.alloc(64, 0);
+  toBuffer(cas).copy(buf);
+  dataBuf.copy(buf, 32);
+  const casBuf = keccak256(buf);
+
+  buf = Buffer.alloc(64, 0);
+  toBuffer(consensus).copy(buf);
+  casBuf.copy(buf, 32);
+  return bufferToHex(keccak256(buf));
+}
+
+const web3 = new Web3();
+const provider = new HDWalletProvider(validatorPriv, providerUrl);
+const claimantAddr = bufferToHex(privateToAddress(validatorPriv));
+web3.setProvider(provider);
+const swapRegistry = new web3.eth.Contract(swapAbi, swapAddr);
+const operatorContract = new web3.eth.Contract(
+  operatorAbi,
+  operatorContractAddr
+);
+
+const run = async function() {
+  let slotId;
+
+  // get events and filter out data
+  const events = await getPastEvents(operatorContract, fromHeight, toHeight);
+  const data = events
+    .filter(event => {
+      const { owner } = event.returnValues;
+      return owner && owner.toLowerCase() === claimantAddr.toLowerCase();
+    })
+    .map(event => {
+      const { owner, blocksRoot } = event.returnValues;
+      slotId = event.returnValues.slotId; // eslint-disable-line prefer-destructuring
+
+      const consenusBuf = Buffer.alloc(64, 0);
+      toBuffer(blocksRoot).copy(consenusBuf);
+      const consensusRoot = bufferToHex(keccak256(consenusBuf));
+
+      const validatorBuf = Buffer.alloc(32, 0);
+      validatorBuf.writeUInt8(slotId, 11);
+      toBuffer(owner).copy(validatorBuf, 12);
+      const validatorData = bufferToHex(validatorBuf);
+
+      const empty = bufferToHex(Buffer.alloc(32, 0));
+
+      return {
+        consensusRoot,
+        cas: empty,
+        validatorData,
+        rest: empty,
+      };
+    })
+    .reduce((accumulator, event) => {
+      let acc = accumulator;
+      if (!acc.push) {
+        acc = [[], [], [], []];
+      }
+      acc[0].push(event.consensusRoot);
+      acc[1].push(event.cas);
+      acc[2].push(event.validatorData);
+      acc[3].push(event.rest);
+      return acc;
+    });
+
+  // read data from bridge
+  const bridgeAddr = await operatorContract.methods.bridge().call();
+  const bridge = new web3.eth.Contract(bridgeABI, bridgeAddr);
+  const startRoot = getPeriodRoot(
+    data[0][0],
+    data[1][0],
+    data[2][0],
+    data[3][0]
+  );
+  const last = data[0].length - 1;
+  const endRoot = getPeriodRoot(
+    data[0][last],
+    data[1][last],
+    data[2][last],
+    data[3][last]
+  );
+  let periodRsp = await bridge.methods.periods(startRoot).call();
+  console.log('start height: ', periodRsp.height);
+  periodRsp = await bridge.methods.periods(endRoot).call();
+  console.log('end height: ', periodRsp.height);
+
+  // claim rewards by submitting
+  const slices = separateForSubmission(data, BATCH_SIZE);
+  let estimate = await swapRegistry.methods
+    .claim(slotId, ...slices[0])
+    .estimateGas({ from: claimantAddr });
+  estimate = Math.floor(estimate * 1.2);
+  for (const slice of slices) {
```

Comment:
`slotId` will be equal to slotId field of the last event here, right? Have you tested this script with several slots? Shouldn't be events grouped by slotId?

Reply:
```es6
const estimate = Math.round(await swapRegistry.methods
    .claim(slotId, ...slices[0])
    .estimateGas({ from: claimantAddr }) * 1.2);
```

Also works

Reply:
> `slotId` will be equal to slotId field of the last event here, right? Have you tested this script with several slots? Shouldn't be events grouped by slotId?

should always ever just work for a single slotId, because of the logic of how we mark slots as claimed: https://github.com/leapdao/leap-contracts/blob/master/contracts/SwapRegistry.sol#L55